pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_drop_database failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_drop_database [failed](https://tanzanite.cluster.engflow.com/invocations/default/cb776ceb-d010-496c-abe7-fb6410e6eaa3?testReportRun=2&testReportShard=31&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on master @ [5f76d93b2ad8d29c14c9b176753ec33cbf81e3fe](https://github.com/cockroachdb/cockroach/commits/5f76d93b2ad8d29c14c9b176753ec33cbf81e3fe):


```
=== RUN   TestTenantLogic_drop_database
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantLogic_drop_database1407365701
    test_log_scope.go:81: use -show-logs to present logs inline
[10:24:23] --- progress: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database: 7 statements
[10:24:25] --- progress: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database: 19 statements
[10:24:27] --- progress: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database: 21 statements
[10:24:31] --- progress: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database: 24 statements
[10:24:34] --- progress: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database: 25 statements
[10:24:37] --- progress: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database: 28 statements
[10:24:39] --- progress: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database: 37 statements
[10:24:42] --- progress: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database: 38 statements
[10:24:45] --- progress: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database: 45 statements
[10:24:47] --- progress: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database: 50 statements
[10:24:49] --- progress: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database: 51 statements
    logic.go:2927: 
         
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database:225: WITH cte AS (
          SELECT job_type, description, status
          FROM crdb_internal.jobs
          WHERE job_type = 'SCHEMA CHANGE' OR job_type = 'SCHEMA CHANGE GC'
          ORDER BY created DESC
          LIMIT 4
        ) SELECT * FROM cte ORDER BY job_type, description
        expected:
            SCHEMA CHANGE     CREATE TABLE constraint_db.public.t2 (t1_id INT8, CONSTRAINT fk FOREIGN KEY (t1_id) REFERENCES constraint_db.public.t1 (a), INDEX (t1_id))  succeeded
            SCHEMA CHANGE     updating referenced FK table t1(125) for table t2(126)                                                                                      succeeded
            SCHEMA CHANGE GC  GC for DROP DATABASE constraint_db CASCADE                                                                                                  running
            SCHEMA CHANGE GC  GC for DROP DATABASE d2 CASCADE                                                                                                             running
            
        but found (query options: "") :
            SCHEMA CHANGE     CREATE TABLE constraint_db.public.t2 (t1_id INT8, CONSTRAINT fk FOREIGN KEY (t1_id) REFERENCES constraint_db.public.t1 (a), INDEX (t1_id))  succeeded
            SCHEMA CHANGE     updating referenced FK table t1(125) for table t2(126)                                                                                      succeeded
            SCHEMA CHANGE GC  GC for DROP DATABASE constraint_db CASCADE                                                                                                  running
            SCHEMA CHANGE GC  GC for DROP DATABASE d2 CASCADE                                                                                                             succeeded
            
[10:24:50] --- done: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database with config 3node-tenant: 52 tests, 1 failures
    logic.go:4005: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database:239: error while processing
    logic.go:4005: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/drop_database:239: too many errors encountered, skipping the rest of the input
    panic.go:541: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogic_drop_database1407365701
--- FAIL: TestTenantLogic_drop_database (76.30s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=31</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_drop_database.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34762

--- a/pkg/sql/logictest/testdata/logic_test/drop_database
+++ b/pkg/sql/logictest/testdata/logic_test/drop_database
@@ -222,19 +222,20 @@ statement ok
 DROP DATABASE constraint_db CASCADE
 
 skipif config local-legacy-schema-changer
-query TTT
+query TT
 WITH cte AS (
-  SELECT job_type, description, status
+  SELECT job_type, description
   FROM crdb_internal.jobs
-  WHERE job_type = 'SCHEMA CHANGE' OR job_type = 'SCHEMA CHANGE GC'
+  WHERE (job_type = 'SCHEMA CHANGE' OR job_type = 'SCHEMA CHANGE GC')
+  AND (status = 'succeeded' OR status = 'running')
   ORDER BY created DESC
   LIMIT 4
 ) SELECT * FROM cte ORDER BY job_type, description
 ----
-SCHEMA CHANGE     CREATE TABLE constraint_db.public.t2 (t1_id INT8, CONSTRAINT fk FOREIGN KEY (t1_id) REFERENCES constraint_db.public.t1 (a), INDEX (t1_id))  succeeded
-SCHEMA CHANGE     updating referenced FK table t1(125) for table t2(126)                                                                                      succeeded
-SCHEMA CHANGE GC  GC for DROP DATABASE constraint_db CASCADE                                                                                                  running
-SCHEMA CHANGE GC  GC for DROP DATABASE d2 CASCADE                                                                                                             running
+SCHEMA CHANGE     CREATE TABLE constraint_db.public.t2 (t1_id INT8, CONSTRAINT fk FOREIGN KEY (t1_id) REFERENCES constraint_db.public.t1 (a), INDEX (t1_id))
+SCHEMA CHANGE     updating referenced FK table t1(125) for table t2(126)
+SCHEMA CHANGE GC  GC for DROP DATABASE constraint_db CASCADE
+SCHEMA CHANGE GC  GC for DROP DATABASE d2 CASCADE
 
 query TTTTTT rowsort
 SHOW DATABASES

